tests/chromium/chromium.spec.ts

```diff
@@ -30,6 +30,118 @@ test('should create a worker from a service worker', async ({ page, server }) =>
   expect(await worker.evaluate(() => self.toString())).toBe('[object ServiceWorkerGlobalScope]');
 });
 
+test('should create a worker from service worker with noop routing', async ({ context, page, server }) => {
```

Comment:
I think we should also verify that `page` does not report the requests sent from within the service worker, only the request being sent from the page (being intercepted by the service-worker)

Reply:
> I think we should also verify that page does not report the requests sent from within the service worker, only the request being sent from the page (being intercepted by the service-worker)

@Meir017 Let me make sure I'm following!

So if `example.html` looks like:

```html
<script>
  navigator.serviceWorker.register('/worker.js');
  navigator.serviceWorker.ready.then(async registration => {
      await fetch('/payload.txt');
  });
</script>
```

and `worker.js` looks like:

```js
fetch('/top-level-sw-request.txt');
self.addEventListener('fetch', event => {
  if (!event.request.url.endsWith('/payload.txt')) return;
  event.respondWith(fetch(event.request));
});
```

On which Playwright object (`page` or `serviceWorker`) would you expect to see the `request` event fire for each of the two requests (`/payload.txt` and `/top-level-sw-request.txt`)?

Reply:
I would expect to see two `request` events fired. one for the page and another for the service worker.
this can be seen in the network tab of the browser.

from your example
- page request `/worker.js`
- sw request `/top-level-sw-request.txt`
- page request `/payload.txt`
- sw request `/payload.txt`

see my comment https://github.com/microsoft/playwright/pull/12565#issuecomment-1061419170